### ISSUE TITLE
Localization issues for CTFL 4 SK

### ISSUE THREAD

**Issue by danopolan** [istqborg/istqb_product_base#51]
There are a few things we would like to get fixed before the release of the CTFL 4 SK version.

1. `Contents` should be translated as `Obsah` (applies for all language versions)
![image](https://github.com/istqborg/istqb_product_base/assets/95688217/3157f38d-f55d-4dcc-9daa-04413f3ded7f)
2. From the description in [Issue #10](https://github.com/istqborg/istqb_product_base/issues/10#issuecomment-2049406254) I a not sure how I can update section reference to support [other grammatical cases](https://github.com/istqborg/istqb-ctfl/blob/main/syllabus-sk/00-introduction.md?plain=1#L63)
![image](https://github.com/istqborg/istqb_product_base/assets/95688217/257c6c74-d08e-4573-843d-c66209a50466)
![image](https://github.com/istqborg/istqb_product_base/assets/95688217/13bcd9a8-4d24-4036-a21d-266e319ad2c5)
3. Table overflow in [section 0.12](https://github.com/istqborg/istqb-ctfl/blob/main/syllabus-sk/00-introduction.md?plain=1#L110)
![image](https://github.com/istqborg/istqb_product_base/assets/95688217/8c3a441f-55bc-4c8b-b385-2718d8bc4435)
4. How can we force a line brake in the case of prepositions?
![image](https://github.com/istqborg/istqb_product_base/assets/95688217/118b1d00-4e15-43d2-9f0e-c681df6afbf6)
![image](https://github.com/istqborg/istqb_product_base/assets/95688217/6b0d43f3-375b-47df-9bf7-c713c56dd92d)
5. Can you please help me with the [appendix B tables](https://github.com/istqborg/istqb-ctfl/blob/main/syllabus-sk/apx-b-traceability-matrix.md?plain=1)? I was not able to put it together in LaTeX code. We would need one exact example to reuse everywhere else.
![image](https://github.com/istqborg/istqb_product_base/assets/95688217/6fb8e973-39f4-440c-b795-21106c938f55)

**Comment by Witiko** [comment]
> 1. `Contents` should be translated as `Obsah` (applies for all language versions)
> ![image](https://github.com/istqborg/istqb_product_base/assets/95688217/3157f38d-f55d-4dcc-9daa-04413f3ded7f)

We currently use the default LaTeX translation for the table of contents. A new field will need to be added to all files `languages/*.yml` in this repository.

> From the description in https://github.com/istqborg/istqb_product_base/issues/10#issuecomment-2049406254 I a not sure how I can update section reference to support [other grammatical cases](https://github.com/istqborg/istqb-ctfl/blob/main/syllabus-sk/00-introduction.md?plain=1#L63)
> ![image](https://github.com/istqborg/istqb_product_base/assets/95688217/257c6c74-d08e-4573-843d-c66209a50466)
> ![image](https://github.com/istqborg/istqb_product_base/assets/95688217/13bcd9a8-4d24-4036-a21d-266e319ad2c5)

This use case is already covered by [the example document](https://github.com/istqborg/istqb_product_base/releases/download/latest/example-document.pdf):

![image](https://github.com/istqborg/istqb_product_base/assets/603082/4a1edd6d-6c9f-49b2-8af7-e282c1633aca)

Namely, in your examples, you would write:

``` markdown
možno nájsť v [sekcii `\ref{section:references}`{=tex}](#section:references)
```
``` markdown
testovania je ďalej vysvetlené v [kapitole `\ref{section:test-planning}`{=tex}](#section:test-planning).
```

If this is a mouthful, we may introduce a nicer syntax, e.g.:

``` markdown
možno nájsť v [sekcii <#section:references>{-}](#section:references)
```
``` markdown
testovania je ďalej vysvetlené v [kapitole <#section:test-planning>{-}](#section:test-planning).
```

Also, perhaps we don't want to capitalize "Sekci" and "Sekciu" by default and would prefer just "sekci" and "sekciu"?

> 3. Table overflow in [section 0.12](https://github.com/istqborg/istqb-ctfl/blob/main/syllabus-sk/00-introduction.md?plain=1#L110)
> ![image](https://github.com/istqborg/istqb_product_base/assets/95688217/8c3a441f-55bc-4c8b-b385-2718d8bc4435)

It's unclear why the table rows would not break. I will review this issue more closely.

> 4. How can we force a line brake in the case of prepositions?
> ![image](https://github.com/istqborg/istqb_product_base/assets/95688217/118b1d00-4e15-43d2-9f0e-c681df6afbf6)
> ![image](https://github.com/istqborg/istqb_product_base/assets/95688217/6b0d43f3-375b-47df-9bf7-c713c56dd92d)

This can happen automatically according to the typographic rules for Czech and Slovak using [the `luavlna` package](https://ctan.org/pkg/luavlna). However, this would require us to switch to the LuaTeX engine, which may result in up to 3× slowdown of the CI according to some measurements. Let me know whether this seems worth a try or whether the current speed of the CI seems already limiting.

Alternatively, you can type [the no-break space Unicode character](https://unicode-explorer.com/c/00A0) (U+00A0, ` `) between the preposition and the following word.

> 5. Can you please help me with the [appendix B tables](https://github.com/istqborg/istqb-ctfl/blob/main/syllabus-sk/apx-b-traceability-matrix.md?plain=1)? I was not able to put it together in LaTeX code. We would need one exact example to reuse everywhere else.
> ![image](https://github.com/istqborg/istqb_product_base/assets/95688217/6fb8e973-39f4-440c-b795-21106c938f55)

Of course. Would you like the LaTeX code for the table in the image?

**Comment by danopolan** [comment]
Ad 1) Please add it.
Ad 2) A shorter version like `[kapitole <#section:references>{-}](#section:references)` would be nice. Also rename `Sekcia` to `kapitola`
Ad 3) Thx
Ad 4) I will add U+00A0 there
Ad 5) LaTeX code please. We will reuse that code for all the syllabi then.

**Comment by Witiko** [comment]
> Also rename `Sekcia` to `kapitola`

In Czech as well?

**Comment by Witiko** [comment]
In PR https://github.com/istqborg/istqb_product_base/pull/56, I fixed issues 1 and 2.

I am failing to reproduce issue 3 with the example document from this repository:

![image](https://github.com/istqborg/istqb_product_base/assets/603082/41e6d41d-92d4-409b-8b33-5676006f3952)

I will now try to reproduce issue 3 directly with the document `syllabus-sk.tex` from the repository `istqborg/istqb-ctfl`.

Issue 4 seems fixed, unless you would be interested in inserting no-break spaces after prepositions automatically. I mentioned a CI slowdown but I have since come up with a solution that would introduce no noticeable slowdown.

I will also produce the code for the table from the screenshot from issue 5 and submit it here in a following comment.

**Comment by Witiko** [comment]
> I will also produce the code for the table from the screenshot from issue 5 and submit it here in a following comment.

@danopolan Here is markdown text that you might use to typeset the table from the screenshot:

``````` md
# Business Outcomes traceability matrix with Learning Objectives {.landscape}

This section lists the number of Foundation Level Learning Objectives related to the Business Outcomes and the traceability between Foundation Level Business Outcomes and Foundation Level Learning Objectives.

``` {=tex}
\begingroup
\renewcommand\tabularxcolumn[1]{m{#1}}  % Vertically center the rows
\renewcommand\arraystretch{1.2}         % Add vertical space around rows
\definecolor{istqbblue}{HTML}{B5C5E5}   % Define the color of cell "Business Outcomes"
\begin{tabularx}{\linewidth}{|m{9mm}|X|c|c|c|c|c|c|c|c|c|c|c|c|c|c|}
   \hline
\multicolumn{2}{|l|}{\cellcolor{istqbblue}\Large Business Outcomes: Foundation Level}
&  \rotatebox[origin=c]{270}{~FL-BO1~}
&  \rotatebox[origin=c]{270}{~FL-BO2~}
&  \rotatebox[origin=c]{270}{~FL-BO3~}
&  \rotatebox[origin=c]{270}{~FL-BO4~}
&  \rotatebox[origin=c]{270}{~FL-BO5~}
&  \rotatebox[origin=c]{270}{~FL-BO6~}
&  \rotatebox[origin=c]{270}{~FL-BO7~}
&  \rotatebox[origin=c]{270}{~FL-BO8~}
&  \rotatebox[origin=c]{270}{~FL-BO9~}
&  \rotatebox[origin=c]{270}{~FL-BO10~}
&  \rotatebox[origin=c]{270}{~FL-BO11~}
&  \rotatebox[origin=c]{270}{~FL-BO12~}
&  \rotatebox[origin=c]{270}{~FL-BO13~}
&  \rotatebox[origin=c]{270}{~FL-BO14~}
\\ \hline
BO1 & Understand what testing is and why it is beneficial & 6 &&&&&&&&&&&&& \\ \hline
BO2 & Understand fundamental concepts of software testing && 22 &&&&&&&&&&&& \\ \hline
BO3 & Identify the test approach and activities to be implemented depending on the context
of testing &&& 6 &&&&&&&&&&& \\ \hline
BO4 & Assess and improve the quality of documentation &&&& 9 &&&&&&&&&& \\ \hline
BO5 & Increase the effectiveness and efficiency of testing &&&&& 20 &&&&&&&&& \\ \hline
BO6 & Align the test process with the software development lifecycle &&&&&& 6 &&&&&&&& \\ \hline
BO7 & Understand test management principles &&&&&&& 6 &&&&&&& \\ \hline
BO8 & Write and communicate clear and understandable defect reports &&&&&&&& 1 &&&&&& \\ \hline
BO9 & Understand the factors that influence the priorities and efforts related to testing &&&&&&&&& 7 &&&&& \\ \hline
BO10 & Works par of a cross-functional &&&&&&&&&& 8 &&&& \\ \hline
BO11 & Know risks and benefits related to test automation &&&&&&&&&&& 1 &&& \\ \hline
BO12 & Identify essential skills required for testing &&&&&&&&&&&& 5 && \\ \hline
BO13 & Understand the impact of risk on testing &&&&&&&&&&&&& 4 & \\ \hline
BO14 & Effectively report on test progress and quality &&&&&&&&&&&&&& 4 \\ \hline
\end{tabularx}
\endgroup
```
```````

Here is how the above code looks when you typeset it with the template:

![image](https://github.com/istqborg/istqb_product_base/assets/603082/db8fe211-79a9-4a07-89a4-dc763e25b42e)

**Comment by Witiko** [comment]
> > 3. Table overflow in [section 0.12](https://github.com/istqborg/istqb-ctfl/blob/main/syllabus-sk/00-introduction.md?plain=1#L110)
>
> I [failed](https://github.com/istqborg/istqb_product_base/assets/603082/41e6d41d-92d4-409b-8b33-5676006f3952) to reproduce issue 3 with the example document from this repository. I will now try to reproduce issue 3 directly with the document `syllabus-sk.tex` from the repository `istqborg/istqb-ctfl`.

After I reviewed the document [`syllabus-sk.pdf`][1] from the repository `istqborg/istqb-ctfl`, I concluded that [the screenshot][2] from the original post seems outdated. Here is how pages 14 and 15 of [`syllabus-sk.pdf`][1] look at the moment:

![image](https://github.com/istqborg/istqb_product_base/assets/603082/8bca13e8-a2bb-47af-afb0-be117f0a1d84)

Therefore, issue 3 seems resolved and PR #56 should close this ticket.

 [1]: https://github.com/istqborg/istqb-ctfl/releases/download/latest/syllabus-sk.pdf
 [2]: https://github.com/istqborg/istqb_product_base/assets/603082/43a50820-eea7-4558-9fe1-b8bf0732cada
